server/ce2e/plugin_test.go

```diff
@@ -314,3 +331,269 @@ func TestMessageHasBeenPostedNewDirectMessageE2E(t *testing.T) {
+	require.NoError(t, err)
+
+	err = mattermost.AddUserToTeam(context.Background(), "notconnected1", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "mmprimary1", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "msteamsprimary1", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "notconnected2", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "mmprimary2", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "msteamsprimary2", "test")
+	require.NoError(t, err)
+	err = mattermost.AddUserToTeam(context.Background(), "msteams_synthetic", "test")
+	require.NoError(t, err)
+
+	notConnected1, _, err := adminClient.GetUserByUsername(context.Background(), "notconnected1", "")
+	require.NoError(t, err)
+	notConnected2, _, err := adminClient.GetUserByUsername(context.Background(), "notconnected2", "")
+	require.NoError(t, err)
+	mmPrimary1, _, err := adminClient.GetUserByUsername(context.Background(), "mmprimary1", "")
+	require.NoError(t, err)
+	mmPrimary2, _, err := adminClient.GetUserByUsername(context.Background(), "mmprimary2", "")
+	require.NoError(t, err)
+	msteamsPrimary1, _, err := adminClient.GetUserByUsername(context.Background(), "msteamsprimary1", "")
+	require.NoError(t, err)
+	msteamsPrimary2, _, err := adminClient.GetUserByUsername(context.Background(), "msteamsprimary2", "")
+	require.NoError(t, err)
+	synthetic, _, err := adminClient.GetUserByUsername(context.Background(), "msteams_synthetic", "")
+	require.NoError(t, err)
+
+	err = store.SetUserInfo(mmPrimary1.Id, "ms-mmprimary1", &fakeToken)
+	require.NoError(t, err)
+	setUserDefaultPlatform(t, mattermost, mmPrimary1, "mattermost")
+	err = store.SetUserInfo(mmPrimary2.Id, "ms-mmprimary2", &fakeToken)
+	require.NoError(t, err)
+	setUserDefaultPlatform(t, mattermost, mmPrimary2, "mattermost")
+	err = store.SetUserInfo(msteamsPrimary1.Id, "ms-msteamsprimary1", &fakeToken)
+	require.NoError(t, err)
+	setUserDefaultPlatform(t, mattermost, msteamsPrimary1, "msteams")
+	err = store.SetUserInfo(msteamsPrimary2.Id, "ms-msteamsprimary2", &fakeToken)
+	require.NoError(t, err)
+	setUserDefaultPlatform(t, mattermost, msteamsPrimary2, "msteams")
+	err = store.SetUserInfo(synthetic.Id, "ms-msteams_synthetic", nil)
+	require.NoError(t, err)
+
+	conn, err := mattermost.PostgresConnection(context.Background())
+	require.NoError(t, err)
+
+	// Mark user as synthetic
+	_, err = conn.Exec("UPDATE Users SET RemoteId = 'syntetic-user' WHERE Id = $1", synthetic.Id)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	ttCases := []struct {
+		name                         string
+		fromUser                     *model.User
+		toUser                       *model.User
+		expectedWithSelectiveSync    bool
+		expectedWithoutSelectiveSync bool
+	}{
+		{
```

Comment:
@asaadmahmood it would be great of you go through this table testing and you see something that we should test here, and if there is anything there that doesn't fit with your expectations.

Also, take into account that this is only "mattermost to msteams", the other way needs to go in another set of tests.